python/tskit/trees.py

```diff
@@ -1981,6 +1981,23 @@ def __next__(self):
         return self.tree
 
 
+class _ReversibleIterator(object):
+    """
+    Simple class to provide a reversible iterator over getters like ts.node() & ts.edge()
+    """
+    def __init__(self, getter, getter_length):
+        self.getter = getter
+        self.getter_length = getter_length
+
+    def __iter__(self):
+        for j in range(self.getter_length):
+            yield self.getter(j)
+
+    def __reversed__(self):
+        for j in reversed(range(self.getter_length)):
```

Comment:
I'm sure it does, but want to be certain: does ``reversed(range())`` do it without making an intermediate list?

Reply:
```
>>> reversed(range(10))
<range_iterator object at 0x10470ac00>
```
So yes. But at the moment the `_ReversibleIterator` doesn't allow `next()`, so we have lost some functionality here...

Reply:
Can you give an example of next not doing the right thing here?

Reply:
Right, I see the problem. You can't use 'yield' here, you have to return the next value from``__next``. This means you have to store the index and direction as state and increment/decrement explicity. See the TreeIterator, for example.

Reply:
So, we need tests to make sure this is following the protocol correctly also.